packages/packagers/js/src/DevPackager.js

```diff
@@ -116,6 +121,20 @@ export class DevPackager {
         wrapped += JSON.stringify(deps);
         wrapped += ']';
 
+        // For node environment @parcel/transformer-js replaces __dirname and __filename calls with
+        // path.join containing placeholders. The packager now replaces those placeholders with
+        // relative paths so that in the end the __dirname/__filename calls resolve to the correct files
+        if (this.bundle.env.isNode()) {
+          const relPath = normalizeSeparators(
+            path.relative(
+              this.bundle.target.distDir,
+              path.dirname(asset.filePath),
```

Comment:
@mischnic Ok, after a bit of a treasure hunt (which was fun!) I managed to put the correct code paths into DevPackager and ScopeHoistingPackager.

How can I do the `meta` optimization you talked about?

Oh, also, I changed it from `projectRoot` to `asset.filePath` as otherwise it wouldn't actually resolve to the correct file

Reply:
I think I understand it. In the node_replacer rust code, when this gets run I should set some boolean. Then in the transformer I can access that boolean and set the `meta`, e.g. like https://github.com/parcel-bundler/parcel/blob/v2/packages/transformers/js/src/JSTransformer.js#L789

Reply:
@mischnic Let me know if I did [`d249b88` (#7727)](https://github.com/parcel-bundler/parcel/pull/7727/commits/d249b880d304dfe4eef50440a7a6cba556c70e41) correctly :)